include/gamepad/recursive_mutex.hpp

```diff
@@ -0,0 +1,27 @@
+#include "pros/apix.h"
+#include "pros/rtos.h"
+#include <system_error>
+
+namespace Gamepad::_impl {
+
```

Comment:
This class needs documentation

Reply:
Addressed in https://github.com/LemLib/Gamepad/pull/6/commits/020af19fdee5829f2d478b4da5b6655a26dffd0d.